sklearn/neighbors/_kde.py

```diff
@@ -181,8 +183,16 @@ def fit(self, X, y=None, sample_weight=None):
 
         algorithm = self._choose_algorithm(self.algorithm, self.metric)
 
-        if self.bandwidth <= 0:
-            raise ValueError("bandwidth must be positive")
+        if self.bandwidth == "scott":
+            self.bandwidth = X.shape[0] ** (-1 / (X.shape[1] + 4))
```

Comment:
You cannot change an input parameter: if this object is fitted twice on data of different shape, the first call will transform the "scott" to a number, and the second call won't implement "scott".

You need to add a "self.bandwidth_" attribute, which stores the numeric value: if given a numeric value, at fit copy self.bandwidth to self.bandwidth_; if given scott of silverman, compute the corresponding value and assign it to self.bandwidth_